openpype/pipeline/colorspace.py

```diff
@@ -0,0 +1,464 @@
+            break
+        if colorspace_key in path:
+            colorspace_name = colorspace_key
+            break
+
+    if not colorspace_name:
+        log.info("No matching colorspace in config '{}' for path: '{}'".format(
+            config_path, path
+        ))
+        return None
+
+    return colorspace_name
+
+
+def validate_imageio_colorspace_in_config(config_path, colorspace_name):
+    """Validator making sure colorspace name is used in config.ocio
+
+    Args:
+        config_path (str): path leading to config.ocio file
+        colorspace_name (str): tested colorspace name
+
+    Raises:
+        KeyError: missing colorspace name
+
+    Returns:
+        bool: True if exists
+    """
+    colorspaces = get_ocio_config_colorspaces(config_path)
+    if colorspace_name not in colorspaces:
+        raise KeyError(
+            "Missing colorspace '{}' in config file '{}'".format(
+                colorspace_name, config_path)
+        )
+    return True
+
+
+def get_ocio_config_colorspaces(config_path):
+    """Get all colorspace data
+
+    Wrapper function for aggregating all names and its families.
+    Families can be used for building menu and submenus in gui.
+
+    Args:
+        config_path (str): path leading to config.ocio file
+
+    Returns:
+        dict: colorspace and family in couple
+    """
+    if sys.version_info[0] == 2:
+        return get_colorspace_data_subprocess(config_path)
+
+    from ..scripts.ocio_wrapper import get_colorspace_data
+    return get_colorspace_data(config_path)
+
+
+def get_colorspace_data_subprocess(config_path):
+    """Get colorspace data via subprocess
+
+    Wrapper for Python 2 hosts.
+
+    Args:
+        config_path (str): path leading to config.ocio file
+
+    Returns:
+        dict: colorspace and family in couple
+    """
+    with _make_temp_json_file() as tmp_json_path:
+        # Prepare subprocess arguments
+        args = [
+            "run", get_ocio_config_script_path(),
+            "config", "get_colorspace",
+            "--in_path", config_path,
+            "--out_path", tmp_json_path
+
+        ]
+        log.info("Executing: {}".format(" ".join(args)))
+
+        process_kwargs = {
+            "logger": log,
+            "env": {}
+        }
+
+        run_openpype_process(*args, **process_kwargs)
+
+        # return all colorspaces
+        return_json_data = open(tmp_json_path).read()
+        return json.loads(return_json_data)
+
+
+def get_ocio_config_views(config_path):
+    """Get all viewer data
+
+    Wrapper function for aggregating all display and related viewers.
+    Key can be used for building gui menu with submenus.
+
+    Args:
+        config_path (str): path leading to config.ocio file
+
+    Returns:
+        dict: `display/viewer` and viewer data
+    """
+    if sys.version_info[0] == 2:
+        return get_views_data_subprocess(config_path)
+
+    from ..scripts.ocio_wrapper import get_views_data
+    return get_views_data(config_path)
```

Comment:
Should both the `get_views_data*` functions be made private? (Prefixed with an underscore). It seems like the intent is to always use `get_ocio_config_view(config_path)`

It's the same with the functions above for `get_colorspace_data` - they shouldn't be as "public" as these functions, no?

Reply:
Honestly not sure how public and private works. From my perspective, this is public because it is imported into another module. But it is private form API perspective, yes.

Reply:
@antirotor @mkolar thoughts on this? 

I'm mostly referring to the `get_views_data_subprocess(config_path)` which is the Py2 wrapper itself. Usually the user shouldn't need to call it explicitly because `get_ocio_config_view` itself decides based on Python version which to use.